crates/proto/src/xfer/dnssec_dns_handle/mod.rs

```diff
@@ -658,13 +673,17 @@ where
     };
 
     // if the DS record was an NSEC then we have an insecure zone
-    if let Some((query, proof)) = error
+    if let Some((query, _proof)) = error
```

Comment:
This doesn't seem intuitive either: if we have an `error.kind().as_nsec()` that has `proof.is_secure()`, we yield a `ProofError::new(Proof::Insecure, ..)`??

Reply:
(see other comment about counter-intuitiveness.)